examples/cluster/plot_inductive_clustering.py

```diff
@@ -28,13 +28,27 @@
 from sklearn.cluster import AgglomerativeClustering
 from sklearn.datasets import make_blobs
 from sklearn.ensemble import RandomForestClassifier
-from sklearn.utils.metaestimators import if_delegate_has_method
+from sklearn.utils.metaestimators import available_if
+from sklearn.utils.validation import check_is_fitted
 
 
 N_SAMPLES = 5000
 RANDOM_STATE = 42
 
 
+def _classifier_has(attr):
+    """Check if we can delegate a method to the underlying classifier.
+
+    First, we check the first fitted classifier if available, otherwise we
+    check the unfitted classifier.
+    """
+    return lambda estimator: (
+        hasattr(estimator.classifier_, attr)
+        if hasattr(estimator, "classifier_")
+        else hasattr(estimator.classifier, attr)
+    )
```

Comment:
Maybe a bit more readable,

```suggestion
def func(estimator):
   clf = getattr(estimator, "classifier_", estimator.classifier)
   return hasattr(clf, attr)

return func
```
bit I'm also fine with the current version. If you prefer the current one, feel free to merge.

Reply:
I think that we applied the same pattern elsewhere so I think this is fine to keep as-is.